addon_service/addon_operation_invocation/views.py

```diff
@@ -20,10 +23,23 @@ def get_permissions(self):
             case "partial_update" | "update" | "destroy":
                 return [IsAuthenticated(), SessionUserIsOwner()]
             case "create":
-                return [IsAuthenticated()]  # additional checks in serializer
+                return [IsAuthenticated(), SessionUserMayPerformInvocation()]
             case None:
                 return super().get_permissions()
             case _:
                 raise NotImplementedError(
                     f"no permission implemented for action '{self.action}'"
                 )
+
+    def perform_create(self, serializer):
+        super().perform_create(serializer)
+        # after creating the AddonOperationInvocation, look into invoking it
+        _invocation = serializer.instance
+        _operation_type = _invocation.operation.operation_type
+        match _operation_type:
+            case AddonOperationType.REDIRECT | AddonOperationType.IMMEDIATE:
+                perform_invocation__blocking(_invocation)
+            case AddonOperationType.EVENTUAL:
+                raise NotImplementedError("TODO: enqueue task")
```

Comment:
Does this mean none of our current addon ops are of `EVENTUAL` type?

Reply:
correct -- the only operations we need at this phase are `IMMEDIATE` (for read-only filetree browsing)

`EVENTUAL` operations may (eventually) include filetree copy/move/delete operations (still handled by waterbutler) and archiving (still handled by osf's archiver and/or pigeon), but we're deferring the complexity of "eventual" state handling until we actually want to use it